adapters/repos/db/lsmkv/compactor_map.go

```diff
@@ -252,3 +264,25 @@ func (c *compactorMap) writeHeader(level, version, secondaryIndices uint16,
 
 	return nil
 }
+
+// WARN: method can alter input slice
+func (c *compactorMap) cleanupValues(values []value) (vals []value, skip bool) {
+	if !c.cleanupTombstones {
+		return values, false
+	}
+
+	last := 0
+	for i := 0; i < len(values); i++ {
+		if !values[i].tombstone {
+			// swap instead overwritting, to prevent multiple (values[x].value)s
+			// pointing to the same slice
```

Comment:
I not sure I understand this part, do you mind elaborating on why this is necessary?

Reply:
updated comments explaining working of `cleanupTombstones` methods.